projects/step-core/src/lib/modules/table/components/custom-columns/custom-columns.component.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  Output,
+  QueryList,
+  SimpleChanges,
+  TrackByFunction,
+  ViewChildren,
+} from '@angular/core';
+import { AugmentedScreenService } from '../../../../client/augmented/step-augmented-client.module';
+import { Input as ColInput } from '../../../../client/generated';
+import { map } from 'rxjs';
+import { MatColumnDef } from '@angular/material/table';
+import { SearchColDirective } from '../../directives/search-col.directive';
+
+@Component({
+  selector: 'step-custom-columns',
+  templateUrl: './custom-columns.component.html',
+  styleUrls: ['./custom-columns.component.scss'],
+})
+export class CustomColumnsComponent implements OnChanges {
+  @Input() screen!: string;
+  @Input() excludeFields?: string[];
+  @Input() isSearchDisabled?: boolean;
+
+  @Output() columnsReady = new EventEmitter<unknown>();
+
+  columns: ColInput[] = [];
+
+  readonly trackByCol: TrackByFunction<ColInput> = (index, item) => item.id;
+
+  @ViewChildren(MatColumnDef) colDef?: QueryList<MatColumnDef>;
+  @ViewChildren(SearchColDirective) searchColDef?: QueryList<SearchColDirective>;
+
+  constructor(private _screens: AugmentedScreenService) {}
+
+  ngOnChanges(changes: SimpleChanges): void {
```

Comment:
Why are you using `ngOnChanges` here? The way this component is used, `this.screen` and `this.excludeFields` are constants and it would not make any sense to ever use variables here? `ngOnInit` would be sufficient:
```
  ngOnInit(): void {
      this.loadColumns(this.screen, this.excludeFields);
  }
```

Reply:
Since these fields are inputs, they can be changed at runtime (event we don't have such cases now).
So I just wish to support the ability to reconfigure custom columns, in case if it's parameters will be changed.

Reply:
Technically I understand, but in this case I can not see a usage for this component having the input changed. But right, maybe there will be a usage in the future, so why not supporting it.. :+1: